src/services/CatalogService.ts

```diff
@@ -31,4 +44,60 @@ export const catalogService: ICatalogService = {
 
     await events.createEvent(eserviceSeedToCreateEvent(eserviceSeed));
   },
+  updateEService: async (
+    eServiceId: string,
+    eservicesSeed: ApiEServiceSeed
+  ): Promise<void> => {
+    const organizationId = await readModelGateway.getOrganizationID();
+    const eservice = await readModelGateway.getEServiceById(eServiceId);
+
+    if (eservice === undefined) {
+      throw eServiceNotFound(eServiceId);
+    }
+
+    if (eservice.producerId !== organizationId) {
+      throw operationForbidden;
+    }
+
+    if (
+      !(
+        eservice.descriptors.length === 0 ||
+        (eservice.descriptors.length === 1 &&
+          eservice.descriptors[0].state === "DRAFT")
+      )
+    ) {
+      throw eServiceCannotBeUpdated(eServiceId);
+    }
+
+    const eserviceSeed = convertToClientEServiceSeed(
+      eservicesSeed,
+      organizationId
+    );
+
+    await events.createEvent({
+      streamId: eServiceId,
+      version: await readModelGateway.getVersion(eServiceId),
+      type: "EServiceUpdated",
+      data: eserviceSeed,
+    });
+  },
+  deleteEService: async (eServiceId: string): Promise<void> => {
+    const organizationId = await readModelGateway.getOrganizationID();
+    const eservice = await readModelGateway.getEServiceById(eServiceId);
+
+    if (eservice === undefined) {
+      throw eServiceNotFound(eServiceId);
+    }
+
+    if (eservice.producerId !== organizationId) {
+      throw operationForbidden;
+    }
+
+    await events.createEvent({
+      streamId: eServiceId,
+      version: await readModelGateway.getVersion(eServiceId),
```

Comment:
It probably makes sense to save the current version together with the entity in the read model so that `readModelGateway.getEServiceById` returns both at the same time (so we know they're always in sync).

Anyway this is fine now, let's think about the read model structure later.